### PR TITLE
Add graphQL, column data controller, click handler for nodes, start Apollo server.

### PR DESCRIPTION
**Pull request by JarodCrawford**
## Description:
This pull request introduces significant updates to the project's data management and user interaction capabilities. It adds GraphQL integration, a new column data controller, and a click handler for nodes. Additionally, it includes the setup to start an Apollo server. These changes aim to enhance the application's efficiency by enabling updates to specific data points without the need to requery the entire database.

## Changes Proposed:
1. **GraphQL Integration:** Implementation of GraphQL to manage queries more efficiently.
2. **Column Data Controller:** Addition of a new controller to manage column data operations.
3. **Click Handler for Nodes:** Implementation of a click handler to interact with individual nodes in the UI.
4. **Apollo Server Setup:** Configuration and initiation of Apollo Server to support GraphQL operations.

## Reasoning:
- **GraphQL Integration:** Reduces the amount of data transferred over the network and improves query efficiency.
- **Column Data Controller:** Enhances the modularity and scalability of data management.
- **Click Handler for Nodes:** Improves user experience by allowing interactive data manipulation.
- **Apollo Server Setup:** Provides a robust and scalable server environment for GraphQL operations.

## Additional Notes:
- The GraphQL schema and resolvers have been aligned with the existing data structure.
- Frontend modifications were made to accommodate the new click handler functionality.

## Testing Strategy:
- Comprehensive unit and integration tests for the GraphQL implementation.
- UI tests to ensure the click handler operates as expected.
- Performance tests to compare data retrieval times with and without GraphQL.
- End-to-end testing of the Apollo server integration.

## Impact Assessment:
- **Performance:** Expected to significantly reduce data loading times and improve user experience.
- **Scalability:** Enhances the application's ability to handle larger datasets and complex queries.
- **Maintainability:** Modular approach makes future enhancements and bug fixes easier to implement.

## Checklist:
- [x] Tests written for all major routes and unknown route handling.
- [x] Code formatting and documentation adhering to established standards.
- [x] Successful local testing and verification of test suite functionality.

## Reviewer Request:
Kindly review the proposed test suite for route integration and provide feedback for any improvements or suggestions. Your insights on the GraphQL implementation and its integration with the existing system would be particularly valuable. Looking forward to your constructive feedback!

### CODE REVIEW

File: src/Flow.tsx
Comment (jennyouk) [inline src/Flow.tsx:1]: Let's research to see if we can put more of this fetch request into middleware

**Review by jennyouk — COMMENTED** [review]
Looks good, will work on clean up

**Review by margarethatch — APPROVED** [review]

**Review by alexpalazzo — COMMENTED** [review]
GraphQL implementation looks good, thanks Jarod!

**Comment by margarethatch** [comment]
Super impressive Jarod. Nice work with both TypeScript and GraphQL.

**Pull request merged**